tokio/tests/task_local.rs

```diff
@@ -31,3 +31,16 @@ async fn local() {
     j2.await.unwrap();
     j3.await.unwrap();
 }
+
+#[cfg(tokio_unstable)]
+#[tokio::test(flavor = "current_thread")]
+async fn test_task_id() {
+    use tokio::task;
+
+    let handle = tokio::spawn(async {
+        let id = task::current_id();
+        println!("current task id: {}", id);
+    });
+
+    handle.await.unwrap();
+}
```

Comment:
also, we could probably test that `current_id()` and `handle.id()` return the same value if we gave the spawned task a oneshot channel, like this:
```suggestion
}

#[cfg(tokio_unstable)]
#[tokio::test(flavor = "current_thread")]
async fn task_ids_match() {
    use tokio::{task, sync::oneshot};
    
    let (tx, rx) = oneshot::channel();
    let handle = tokio::spawn(async {
        let id = rx.await.unwrap();
        assert_eq!(id, task::current_id());
    });

    tx.send(handle.id());

    handle.await.unwrap();
}
```